### PR TITLE
eni: Add trigger metrics

### PR DESCRIPTION
**Pull request by tgraf**
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/8521)

### CODE REVIEW

**Comment by tgraf** [comment]
test-me-please

File: pkg/metrics/metrics.go [hunk 1]
```diff
@@ -969,6 +987,39 @@ func CreateConfiguration(metricsEnabled []string) (Configuration, []prometheus.C
...
+			})
+
+			collectors = append(collectors, TriggerPolicyUpdateFolds)
+			c.TriggerPolicyUpdateFolds = true
+
+		case Namespace + "_triggers_policy_update_call_duration_seconds":
+			TriggerPolicyUpdateCallDuration = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+				Namespace: Namespace,
+				Subsystem: "triggers",
+				Name:      "policy_update__call_duration_seconds",
```
Comment (aanm) [inline pkg/metrics/metrics.go]: s/`policy_update__call_duration_seconds`/`policy_update_call_duration_seconds`/

**Review by aanm — CHANGES_REQUESTED** [review]

**Comment by tgraf** [comment]
test-me-please

File: pkg/metrics/metrics.go [hunk 2]
```diff
@@ -969,6 +987,39 @@ func CreateConfiguration(metricsEnabled []string) (Configuration, []prometheus.C
 
 			collectors = append(collectors, BPFMapOps)
 			c.BPFMapOps = true
+
+		case Namespace + "_triggers_policy_update_total":
+			TriggerPolicyUpdateTotal = prometheus.NewCounterVec(prometheus.CounterOpts{
+				Namespace: Namespace,
+				Subsystem: "triggers",
+				Name:      "policy_update_total",
+				Help:      "Total number of policy update trigger invocations labelled by reason",
```
Comment (ianvernon) [inline pkg/metrics/metrics.go]: Nit (non-blocking): labelled --> labeled

File: pkg/metrics/metrics.go [hunk 3]
```diff
@@ -969,6 +987,39 @@ func CreateConfiguration(metricsEnabled []string) (Configuration, []prometheus.C
 
 			collectors = append(collectors, BPFMapOps)
 			c.BPFMapOps = true
+
+		case Namespace + "_triggers_policy_update_total":
+			TriggerPolicyUpdateTotal = prometheus.NewCounterVec(prometheus.CounterOpts{
+				Namespace: Namespace,
+				Subsystem: "triggers",
```
Comment (ianvernon) [inline pkg/metrics/metrics.go]: Make "triggers" a const string.

File: pkg/metrics/metrics.go [hunk 4]
```diff
@@ -969,6 +987,39 @@ func CreateConfiguration(metricsEnabled []string) (Configuration, []prometheus.C
...
+				Name:      "policy_update_folds",
+				Help:      "Current number of folds",
+			})
+
+			collectors = append(collectors, TriggerPolicyUpdateFolds)
+			c.TriggerPolicyUpdateFolds = true
+
+		case Namespace + "_triggers_policy_update_call_duration_seconds":
```
Comment (ianvernon) [inline pkg/metrics/metrics.go]: Could make "triggers" a const string, then the name of each metric a const string and be able to use them in the case statements + the "Subsystem" and "Name" fields for each trigger to reduce duplicate hardcoded strings.

**Review by ianvernon — APPROVED** [review]
Nothing blocking, just some suggestions to reduce hardcoding in metrics string fields. This can be merged.

**Review by joestringer — APPROVED** [review]
Metrics LGTM

**Comment by tgraf** [comment]
test-me-please

**Pull request merged**